tests/integration/test_session.py

```diff
@@ -19,3 +21,27 @@ def _two_factor_auth():
                                         match_requests_on=match):
             r = self.session.get('https://api.github.com/users/sigmavirus24')
             assert r.status_code == 200
+
+    @pytest.fixture(autouse=True)
```

Comment:
Is this the correct way to use this? I'm not familar with pytest but this doesn't seem to use it for only this test and has some side effects in breaking the other test in this class.

Reply:
No. `@pytest.fixture` declares the function as a fixture. `autouse=True` tells it to run before every test function. It seems like you wanted a temporary directory fixture?